### PR TITLE
Force chat scroll after render

### PR DESCRIPTION
**Pull request by sirugh**
So I'm not too familiar with ES6 or React yet, but I saw this issue and it seemed relatively simple to fix.
~~My solution involves getting the DOMNode which is the chat element, then the first child of that which is the messages div, and the last child of that which is the latest message to be received.

This could probably be shortcut if we have a handle on the message div at some point.

I thought it may be in render, but then saw that it is rendering the entirety of the messages after every message.~~

Using refs to grab the messages div and scroll it.

https://github.com/aeosynth/draft/issues/103

### CODE REVIEW

File: public/src/components/chat.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ export default React.createClass({
   hear(msg) {
     this.state.messages.push(msg)
     this.forceUpdate()
+    this.getDOMNode().firstChild.lastChild.scrollIntoView()
```
Comment (sirugh) [inline public/src/components/chat.js]: Alternatively
```document.getElementById('messages').lastChild.scrollIntoView()```

**Comment by aeosynth** [comment]
use `scrollTop = scrollHeight` instead, and wrap it up as an argument to `forceUpdate`, which will ensure that it gets executed once the dom has actually updated

http://facebook.github.io/react/docs/component-api.html#forceupdate

File: public/src/components/chat.js [hunk 2]
```diff
@@ -28,7 +28,11 @@ export default React.createClass({
 
   hear(msg) {
     this.state.messages.push(msg)
-    this.forceUpdate()
+    this.forceUpdate(this.scrollChat)
+  },
+  scrollChat() {
+    let el = document.getElementById('messages')
```
Comment (aeosynth) [inline public/src/components/chat.js]: use refs

**Comment by sirugh** [comment]
Thanks for the hint about refs, @aeosynth, I wasn't aware you could just add the refs attr. Comment addressed!

**Comment by aeosynth** [comment]
merged - https://github.com/aeosynth/draft/commit/8e10dd9c10961623150343783ed846d732d06afe

**Pull request closed without merging**